src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -421,6 +428,65 @@ private static VersionInfo GetVersionInfo()
         }
     }
 
+    /// <summary>
+    /// Base class for the <see cref="ISingleFeaturePredictionTransformer{TModel}"/> working on clustering tasks.
+    /// </summary>
+    /// <typeparam name="TModel">An implementation of the <see cref="IPredictorProducing{TResult}"/></typeparam>
+    public sealed class ClusteringPredictionTransformer<TModel> : SingleFeaturePredictionTransformerBase<TModel>
```

Comment:
>ClusteringPredictionTransformer [](start = 24, length = 31)

I still don't like not making those generic on the scorer as well...

Reply:
addressed on #996

---
In reply to: [219580106](https://github.com/dotnet/machinelearning/pull/979#discussion_r219580106) [](ancestors = 219580106)